packages/wdio-allure-reporter/src/reporter.ts

```diff
@@ -505,88 +573,135 @@ export default class AllureReporter extends WDIOReporter {
+                    if (!this._state.currentTest) {
+                        this.onTestStart(Object.assign({}, hook, { title: hook.currentTest }) as TestStats)
+                    }
+                    // force broken status when hook fails
+                    if (hook.error){
+                        this._endTest(hook.error ? this._options.testStatusFailedHook || AllureStatus.BROKEN : AllureStatus.PASSED, hook.error)
+                    }
+                } else {
+                    // should not pop test case if no steps and before hook (put them back to the list)
+                    if (currentHookRoot){
+                        this._state.push(currentHookRoot)
+                    }
+                    if (currentHookRootStep) {
+                        this._state.push(currentHookRootStep)
+                    }
+                }
             }
-            return
-        }
-
-        if (hook.error) {
-            // add hook as test to suite for mocha all hooks, when it didn't start before
-            if (disableMochaHooks && isMochaAllHook) {
-                this.onTestStart(hook)
+        } else if (!(isAllHook || isEachHook) && !useCucumberStepReporter) {
+            // getting the lates element
+            const lastElement = this._state.pop()
+            if (lastElement) {
+                const isCustomHook = lastElement instanceof AllureTest && lastElement.wrappedItem.name?.startsWith('hook:')
+                this._state.push(lastElement)
+                if (isCustomHook) {
+                    // we end the test case that represent the custom hook call
+                    this._endTest(hook.error ? this._options.testStatusFailedHook || AllureStatus.SKIPPED : AllureStatus.PASSED, hook.error)
+                }
             }
-
-            this.onTestFail(hook)
-            // mocha doesn't fire onTestEnd for failed beforeEach hook
-            // we have to force mark test as failed test
-            if (isMochaBeforeEachHook(hook.title) && !disableMochaHooks) {
-                this.onTestFail(Object.assign({}, hook, { title: hook.currentTest }) as TestStats)
+        } else if (useCucumberStepReporter) {
+            hook.error ? this.onTestFail(hook) : this.onTestPass()
+        } else if (disableHooks && hook.error) { // only if the hook fails, it will be reported
+            const hookExecutable = isBeforeTypeHook(hook.title) ? this._state.currentSuite.addBefore() : this._state.currentSuite.addAfter()
+            const hookStep = hookExecutable.startStep(hook.title)
+
+            // to cover the before/after all hook.
+            if (!this._state.currentTest) {
+                this.onTestStart(Object.assign({}, hook, { title: hook.currentTest }) as TestStats)
             }
 
-            return
+            // set error information
+            hookExecutable.stage = hookStep.stage = Stage.FINISHED
+            hookExecutable.detailsMessage = hookStep.detailsMessage = hook.error.message
```

Comment:
The error object must pass through `getErrorFromFailedTest(err)' to properly format errors from different test runners.